### PR TITLE
Stac 9091 view script api

### PR DESCRIPTION
**Pull request by konstantintatarnikov**
Docs for: 
- view script api
- thenInject combinator

### CODE REVIEW

File: develop/scripting/async_script_result.md [hunk 1]
```diff
@@ -61,6 +61,27 @@ ScriptApi.asyncFn1().then  {
 
 will return an array of both the result of `asyncFn2` and `asyncFn3`.
 
+### Reducing
```
Comment (lowdewijk) [inline develop/scripting/async_script_result.md:64]: Suggestion: "Reducing with `thenInject`"

File: develop/scripting/script-apis/view.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+title: Script API - View
+kind: Documentation
+description: Fetch a list of views.
+---
+
+# Script API: View
+
+## Function `getAll`
+
+Returns query view responses. The query view response contains
```
Comment (lowdewijk) [inline develop/scripting/script-apis/view.md:11]: Suggestion: "Returns a list of all views".

Unfortunately I see that the code does not actually filter the list of views based on the user permissions. So I would keep the description simple and I've made a security-issue for this: https://stackstate.atlassian.net/browse/STAC-9981

File: develop/scripting/script-apis/view.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+* `view` - the actual query view definition
```
Comment (lowdewijk) [inline develop/scripting/script-apis/view.md:13]: The following is a description of the return value. In other script API descriptions we have a separate heading where the return type is described. Please keep that format.

File: develop/scripting/script-apis/view.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+* query - STQL query which can be used subsequently in getting topology using [Topology Script Api](topology.md)
+* state - View state object, holding view state
+
+**Args:**
+
+None
+
+**Builder methods**
+
+`withStarCount` - the flag asking the api to include the star count to the response.
```
Comment (lowdewijk) [inline develop/scripting/script-apis/view.md:45]: Suggestion: adds the number of times a star was given by any user to the view info of each view.

Comment (lowdewijk) [reply develop/scripting/script-apis/view.md:45]: Tip: Try to not describe things in their technical terms, but in terms of user expectations (input/output).

File: develop/scripting/script-apis/view.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+If this flag is set the query response will contain the `viewInfo` holding star count. The stars count is available on this path `viewResponse.viewInfo.stars.count`
+
+**Examples:**
+
+The example showing how to query views and subsequently the names of topology components from those views is given below:
```
Comment (lowdewijk) [inline develop/scripting/script-apis/view.md:51]: Suggestion: The following example collects the names of all components in all views.

File: develop/scripting/script-apis/view.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+```text
+View.getAll()
+    .thenInject([]) { components, viewResponse ->
+      Topology.query(viewResponse.view.query)
+              .components()
+              .then { topologyComponents ->
+                  def currentBatch = topologyComponents.collect { it.name }
```
Comment (lowdewijk) [inline develop/scripting/script-apis/view.md:59]: suggestion to rename `currentBatch` to `componentNames`

File: develop/scripting/script-apis/view.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+If this flag is set the query response will contain the `viewInfo` holding star count. The stars count is available on this path `viewResponse.viewInfo.stars.count`
+
+**Examples:**
+
+The example showing how to query views and subsequently the names of topology components from those views is given below:
+
+```text
+View.getAll()
+    .thenInject([]) { components, viewResponse ->
```
Comment (lowdewijk) [inline develop/scripting/script-apis/view.md:55]: Suggestion to rename: `components` -> `componentNames` and `viewResponse` to `view`.

File: develop/scripting/script-apis/view.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      Topology.query(viewResponse.view.query)
+              .components()
+              .then { topologyComponents ->
+                  def currentBatch = topologyComponents.collect { it.name }
+                  (components + currentBatch).unique()
```
Comment (lowdewijk) [inline develop/scripting/script-apis/view.md:60]: Suggestions: `componentNames + components.collect { it.name }`. Also suggestion to do unique as a separate step at the end.

**Comment by lowdewijk** [comment]
Could you also describe `Async.sequence`? Then we are 100% complete in terms of script API descriptions.

File: develop/scripting/async_script_result.md (hunk 1 above)
Comment (konstantintatarnikov) [reply develop/scripting/async_script_result.md:64]: done

File: develop/scripting/script-apis/view.md (hunk 2 above)
Comment (konstantintatarnikov) [reply develop/scripting/script-apis/view.md:11]: fixed as suggested.

File: develop/scripting/script-apis/view.md (hunk 3 above)
Comment (konstantintatarnikov) [reply develop/scripting/script-apis/view.md:13]: fixed as suggested

File: develop/scripting/script-apis/view.md (hunk 5 above)
Comment (konstantintatarnikov) [reply develop/scripting/script-apis/view.md:51]: fixed as suggested

File: develop/scripting/script-apis/view.md (hunk 8 above)
Comment (konstantintatarnikov) [reply develop/scripting/script-apis/view.md:60]: fixed as suggested

File: develop/scripting/script-apis/view.md (hunk 4 above)
Comment (konstantintatarnikov) [reply develop/scripting/script-apis/view.md]: fixed as suggested

File: develop/scripting/script-apis/view.md (hunk 7 above)
Comment (konstantintatarnikov) [reply develop/scripting/script-apis/view.md]: fixed as suggested

**Pull request merged**